hang of dc_mlx5/test_uct_perf.envelope/0 on swx-rdmz-ucx-new-01

```
2020-04-02T16:08:19.2732179Z [----------] 1 test from dc_mlx5/test_uct_perf
2020-04-02T16:08:19.2732838Z [ RUN      ] dc_mlx5/test_uct_perf.envelope/0 <dc_mlx5/mlx5_3:1>
2020-04-02T16:08:19.5773649Z [     INFO ] mlx5_3:1                am latency : 1.133 usec (performance not checked)
2020-04-02T16:08:19.9484702Z [     INFO ] mlx5_3:1                   am rate : 6.943 Mpps (performance not checked)
2020-04-02T16:08:20.3455565Z [     INFO ] mlx5_3:1                 am rate64 : 6.355 Mpps (performance not checked)
2020-04-02T16:08:20.7512557Z [     INFO ] mlx5_3:1          am bcopy latency : 1.584 usec (performance not checked)
2020-04-02T16:08:20.8467715Z [     INFO ] mlx5_3:1               am bcopy bw : 2472.585 MB/sec (performance not checked)
2020-04-02T16:08:20.9448192Z [     INFO ] mlx5_3:1               am zcopy bw : 2296.791 MB/sec (performance not checked)
2020-04-02T16:08:21.2428396Z [     INFO ] mlx5_3:1               put latency : 1.107 usec (performance not checked)
2020-04-02T16:08:21.6017107Z [     INFO ] mlx5_3:1                  put rate : 7.138 Mpps (performance not checked)
2020-04-02T16:08:21.6891670Z [     INFO ] mlx5_3:1              put bcopy bw : 6257.209 MB/sec (performance not checked)
2020-04-02T16:08:21.7886657Z [     INFO ] mlx5_3:1              put zcopy bw : 4437.511 MB/sec (performance not checked)
2020-04-02T16:08:22.2067110Z [     INFO ] mlx5_3:1        atomic add latency : 1.446 usec (performance not checked)
2020-04-02T16:08:23.6514765Z [     INFO ] mlx5_3:1           atomic add rate : 1.574 Mpps (performance not checked)
2020-04-02T16:08:23.8812067Z [     INFO ] mlx5_3:1       atomic fadd latency : 1.622 usec (performance not checked)
2020-04-02T16:08:24.1136141Z [     INFO ] mlx5_3:1      atomic cswap latency : 1.656 usec (performance not checked)
2020-04-02T16:08:24.3530479Z [     INFO ] mlx5_3:1       atomic swap latency : 1.739 usec (performance not checked)
2020-04-02T16:23:19.2853584Z /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test_helpers.cc:49: Failure
2020-04-02T16:23:19.2856368Z Failed
2020-04-02T16:23:19.2857207Z Connection timed out - abort testing
2020-04-02T16:23:19.2858186Z [swx-rdmz-ucx-new-01:15637:0:15637] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
2020-04-02T16:23:20.6516153Z ==== backtrace (tid:  15637) ====
2020-04-02T16:23:20.6518542Z  0 0x00000000000589b8 ucs_debug_print_backtrace()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../src/ucs/debug/debug.c:656
2020-04-02T16:23:20.6519594Z  1 0x0000000000008f47 pthread_join()  ???:0
2020-04-02T16:23:20.6521078Z  2 0x0000000000594af8 test_perf::run_multi_threaded()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test_perf.cc:244
2020-04-02T16:23:20.6523183Z  3 0x00000000005951d0 test_perf::run_test()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test_perf.cc:270
2020-04-02T16:23:20.6525770Z  4 0x0000000000665f93 test_uct_perf_envelope_Test::test_body()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/uct/test_uct_perf.cc:170
2020-04-02T16:23:20.6527658Z  5 0x0000000000598816 ucs::test_base::run()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test.cc:290
2020-04-02T16:23:20.6529446Z  6 0x0000000000598816 ucs::test_base::TestBodyProxy()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test.cc:316
2020-04-02T16:23:20.6531499Z  7 0x000000000057cea9 HandleSehExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/gtest-all.cc:3562
2020-04-02T16:23:20.6534040Z  8 0x00000000005710ed testing::Test::Run()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/gtest-all.cc:3635
2020-04-02T16:23:20.6536075Z  9 0x00000000005711c5 testing::TestInfo::Run()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/gtest-all.cc:3812
2020-04-02T16:23:20.6537935Z 10 0x000000000057132f testing::TestCase::Run()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/gtest-all.cc:3930
2020-04-02T16:23:20.6539886Z 11 0x0000000000575add testing::internal::UnitTestImpl::RunAllTests()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/gtest-all.cc:5808
2020-04-02T16:23:20.6542128Z 12 0x0000000000575dc0 testing::internal::UnitTestImpl::RunAllTests()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/gtest-all.cc:5725
2020-04-02T16:23:20.6544060Z 13 0x0000000000515e28 RUN_ALL_TESTS()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/gtest.h:20059
2020-04-02T16:23:20.6545836Z 14 0x0000000000515e28 main()  /scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/main.cc:102
2020-04-02T16:23:20.6546785Z 15 0x00000000000223d5 __libc_start_main()  ???:0
2020-04-02T16:23:20.6547428Z 16 0x000000000055b45f _start()  ???:0
2020-04-02T16:23:20.6548031Z =================================
2020-04-02T16:23:20.6549021Z [swx-rdmz-ucx-new-01:15637:0:15637] Process frozen...
2020-04-02T18:47:43.7579794Z make: *** [test] Terminated
```
https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/5194/logs/135

--- a/src/tools/perf/lib/libperf_int.h
+++ b/src/tools/perf/lib/libperf_int.h
@@ -168,10 +168,12 @@ static inline void ucx_perf_get_time(ucx_perf_context_t *perf)
     perf->current.time_acc = ucs_get_accurate_time();
 }
 
-static inline void ucx_perf_omp_barrier()
+static inline void ucx_perf_omp_barrier(ucx_perf_context_t *perf)
 {
 #if _OPENMP
+    if (perf->params.thread_count > 1) {
 #pragma omp barrier
+    }
 #endif
 }
 

--- a/src/tools/perf/lib/ucp_tests.cc
+++ b/src/tools/perf/lib/ucp_tests.cc
@@ -332,7 +332,7 @@ public:
 
         ucx_perf_test_start_clock(&m_perf);
 
-        ucx_perf_omp_barrier();
+        ucx_perf_omp_barrier(&m_perf);
 
         send_buffer   = m_perf.send_buffer;
         recv_buffer   = m_perf.recv_buffer;
@@ -369,7 +369,7 @@ public:
         wait_window(m_max_outstanding, true);
         ucp_worker_flush(m_perf.ucp.worker);
 
-        ucx_perf_omp_barrier();
+        ucx_perf_omp_barrier(&m_perf);
 
         ucx_perf_get_time(&m_perf);
         ucp_perf_barrier(&m_perf);
@@ -399,7 +399,7 @@ public:
 
         ucx_perf_test_start_clock(&m_perf);
 
-        ucx_perf_omp_barrier();
+        ucx_perf_omp_barrier(&m_perf);
 
         send_buffer   = m_perf.send_buffer;
         recv_buffer   = m_perf.recv_buffer;
@@ -435,7 +435,7 @@ public:
         wait_window(m_max_outstanding, true);
         ucp_worker_flush(m_perf.ucp.worker);
 
-        ucx_perf_omp_barrier();
+        ucx_perf_omp_barrier(&m_perf);
 
         ucx_perf_get_time(&m_perf);
 

--- a/test/gtest/ucp/test_ucp_perf.cc
+++ b/test/gtest/ucp/test_ucp_perf.cc
@@ -67,7 +67,7 @@ const test_perf::test_spec test_ucp_perf::tests[] =
 
   { "tag sync mr", "Mpps",
     UCX_PERF_API_UCP, UCX_PERF_CMD_TAG_SYNC, UCX_PERF_TEST_TYPE_STREAM_UNI,
-    UCP_PERF_DATATYPE_CONTIG, 0, 1, { 8 }, 1, 2000000lu,
+    UCP_PERF_DATATYPE_CONTIG, 0, 1, { 8 }, 1, 200000lu,
     ucs_offsetof(ucx_perf_result_t, msgrate.total_average), 1e-6, 0.05, 100.0, 0},
 
   { "tag wild mr", "Mpps",

